CentralComputing/SafeQueue.hpp

```diff
@@ -20,14 +20,15 @@ class SafeQueue {
   * Dequeues the object
   * @return the oldest object on the queue or nullptr if the queue is empty
   */
-  T dequeue() {
+  
+  bool dequeue(T & val) {
```

Comment:
Will address everything else soon, but the specific reason I made this change is that the get() wrapper function in command.cpp explicitly returns a uint16_t or our struct depending on implementation, so having the safeQueue return std::null_ptr would cause a compilation error since it can't be type-casted to uint16_t

Reply:
I agree that the change you made is appropriate. 
Technically you could make it work the original way, it just requires casting,  but that feels uglier